### DISCUSSION TITLE
Viability of unusual directionality (e.g. RTL) in General Information Exchange

### DISCUSSION THREAD

**Discussion by MasterInQuestion** [MasterInQuestion/talk#2] [Linguistic]
Follow-Up: [ https://github.com/microsoft/vscode/issues/11770#issuecomment-1361202820](https://github.com/microsoft/vscode/issues/11770#issuecomment-1361202820)

    Relevant demo: [ https://github.com/MasterInQuestion/attach/commit/e5ddd9778eccefbb394601c0e17b872c6a686902](https://github.com/MasterInQuestion/attach/commit/e5ddd9778eccefbb394601c0e17b872c6a686902)

**Comment by MasterInQuestion** [comment]
It appears Tibetan is also sort of RTL property (perhaps even directionality-free like Chinese), despite being asserted LTR.

    https://earlytibet.com/wp-content/uploads/2014/02/itj629_colophon.jpg
    https://earlytibet.com/2014/02/04/the-original-bodhicaryavatara/
    .
    https://www.pinterest.com/pin/647462883949159168
    https://inkessential.blogspot.com/2009/03/seal-script.html
    https://babelstone.co.uk/Phags-pa/Overview.html#Tibetan

    https://r12a.github.io/scripts/tibt/bo.html#direction
    https://r12a.github.io/scripts/tutorial/summaries/tibetan

**Comment by MasterInQuestion** [comment]
[ Quote iAhmadKarmola @ CE 2023-11-15 00:34:18 UTC:
[ https://github.com/microsoft/vscode/issues/11770#issuecomment-1811611927](https://github.com/microsoft/vscode/issues/11770#issuecomment-1811611927)
    <img src="https://github.com/MasterInQuestion/attach/raw/main/MasterInQuestion/talk/2/BiDi_fail.webp" alt="[ &quot;BiDi_fail.webp&quot; ]" /> ]
<^>    I sincerely couldn't understand how could such be even considered the acceptable working state...
    Tricky seeking, in particular after soft-wrap.

**Reply by MasterInQuestion** [reply]
Hi, @iAhmadKarmola.
    Would you mind giving some comments on this? (merely for linguistic research)
    Thanks.

**Reply by iAhmadKarmola** [reply]
Hello, I did not understand what you mean.
    If you mean that you want documents to support RTL languages in VS Code or documents to be completely right-to-left, I answered in the comments below:
    [https://github<i>.</i>com/microsoft/vscode/issues/11770#issuecomment-1811611927](https://github.com/microsoft/vscode/issues/11770#issuecomment-1811611927)
    [https://github<i>.</i>com/microsoft/vscode/issues/11770#issuecomment-1813469645](https://github.com/microsoft/vscode/issues/11770#issuecomment-1813469645)

**Reply by MasterInQuestion** [reply]
Thanks anyway for your participation.
    The title of this thread much says all. (please trace the links also)

    In brief, how do you read such BiDi (Bi-Directional) content?
    Do you really think current BiDi implementation of things acceptable..?

**Comment by MasterInQuestion** [comment]
I have derived a set of typical RTL characters (APH: Arabic, Persian, Hebrew) practically in use:
    Based on the referential keyboard implementations on Google Translate:
    |&#x2A;| https://translate.google.com/?hl=en&sl=ar
    |&#x2A;| https://translate.google.com/?hl=en&sl=fa
    |&#x2A;| https://translate.google.com/?hl=en&sl=iw
\
    /<a href="https://regex101.com/?regex=%5Bא-תء-غـ-ْ٠-٩ٵپچژکگی۰-۹%5D&flags=g&flavor=javascript">[א-תء-غـ-ْ٠-٩ٵپچژکگی۰-۹]</a>/:
[[
א
ב
ג
ד
ה
ו
ז
ח
ט
י
ך
כ
ל
ם
מ
ן
נ
ס
ע
ף
פ
ץ
צ
ק
ר
ש
ת
ء
آ
أ
ؤ
إ
ئ
ا
ب
ة
ت
ث
ج
ح
خ
د
ذ
ر
ز
س
ش
ص
ض
ط
ظ
ع
غ
ـ
ف
ق
ك
ل
م
ن
ه
و
ى
ي
ً
ٌ
ٍ
َ
ُ
ِ
ّ
ْ
٠
١
٢
٣
٤
٥
٦
٧
٨
٩
ٵ
پ
چ
ژ
ک
گ
ی
۰
۱
۲
۳
۴
۵
۶
۷
۸
۹
]]

    And among which we have the "French F and English F"...
    https://github.com/MasterInQuestion/Markup/blob/main/AAA.htm

    See also:
    |&#x2A;| https://unicode.org/charts/nameslist/n_FE70.html
    |&#x2A;| https://unicode.org/charts/nameslist/n_FB50.html

**Comment by mohmmadView** [comment]
[https://github<i>.</i>com/microsoft/vscode/issues/11770#issuecomment-1987347003](https://github.com/microsoft/vscode/issues/11770#issuecomment-1987347003)

**Reply by MasterInQuestion** [reply]
[https://github<i>.</i>com/microsoft/vscode/issues/11770#issuecomment-1989415954](https://github.com/microsoft/vscode/issues/11770#issuecomment-1989415954)

**Comment by MasterInQuestion** [comment]
@mohmmadView,
    I'm interested in how RTL languages like Arabic are supposed to be parsed, when interleaving LTR content.
    (e.g. natively, the numeral digits; and English etc.)

    Would you mind providing some insights? Thanks.
